### DISCUSSION TITLE
Multiple samplers for one study

### DISCUSSION THREAD

**Discussion by lkaino** [optuna/optuna#5753] [Q&A]
Is it possible to switch to another sampler during a study?

Let's say I would start with 1000 iterations of RandomSampler, after which I would switch to TPESampler for example.

Would it even be possible to run these samplers at the same time?

**Comment by nzw0301** [comment]
Yes, indeed TPESampler does internally use random sampler during the first `n_startup_trials` trials. In your example, just set `n_startup_trials=1000` as an argument of TPESampler to do so.

**Reply by rodrigoscoelho** [reply]
How does the AutoSampler work? Specifically, how many n_startup_trials does it utilize? Any insights would be appreciated—thank you!

**Reply by YYYC-hello** [reply]
I encountered a problem where I set the value of n_startup_trials to 10, which resulted in the parameters being exactly the same for the first ten experiments. Further experiments were conducted using randomSampler, and it was found that the same issue also occurred. May I ask if you can solve this problem?

**Reply by nabenabe0928** [reply]
@YYYC-hello 
We need to know more details:
- What sampler did you pass to `create_study`?
- How many parallel workers do you have? (Is it multi-processing, multi-threading, or a distributed setup?)
- How does your search space look? Or the console log would be very helpful as well.
